core/src/main/scala/com/avast/clients/storage/compression/ZstdDecompressOutputStream.scala

```diff
@@ -0,0 +1,52 @@
+package com.avast.clients.storage.compression
+
+import com.github.luben.zstd.{ZstdDecompressCtx, ZstdInputStream}
+
+import java.io.OutputStream
+import java.nio.ByteBuffer
+import java.nio.channels.Channels
+
+class ZstdDecompressOutputStream(outputStream: OutputStream) extends OutputStream {
+  private val decompressCtx = new ZstdDecompressCtx()
+  private val outputChannel = Channels.newChannel(outputStream)
+  private val closed = false
+
+  override def write(chunk: Array[Byte]): Unit = {
+    if (closed) {
+      throw new IllegalStateException("Stream is closed")
+    }
+
+    val inputBuffer = ByteBuffer.allocateDirect(chunk.length)
```

Comment:
Why not `ByteBuffer.wrap`? 🤔

Reply:
It doesn't work. I've tested that. The `zstd-jni` uses a native (C) library under the hood, and because of that, it needs direct byte buffers.

> Direct vs. non-direct buffers
> A byte buffer is either direct or non-direct. Given a direct byte buffer, the Java virtual machine will make a best effort to perform native I/O operations directly upon it. That is, it will attempt to avoid copying the buffer's content to (or from) an intermediate buffer before (or after) each invocation of one of the underlying operating system's native I/O operations.

Reply:
Ah, ok. Some comment would be useful then :)

Reply:
I've just added a comment.